app/uk/gov/hmrc/trackingconsentfrontend/assets/src/ui/renderBanner.ts

```diff
@@ -5,10 +5,11 @@ import { COOKIE_BANNER_QUESTION_CLASS, COOKIE_BANNER_CLASS, ACCEPT_ALL_CLASS } f
 import removeElement from '../common/removeElement';
 import callIfNotNull from '../common/callIfNotNull';
 
-const handleAcceptAllClick = userPreference => event => {
+const handleAcceptAllClick = (userPreference, preferenceCommunicator) => event => {
     event.preventDefault()
 
     userPreference.userAcceptsAll()
+    preferenceCommunicator.sendPreferences(userPreference)
```

Comment:
It feels like this call should belong in the business logic layer - i.e. the userAcceptsAll method should be responsible for notifying GTM (possibly other systems in future - could adopt a pub/sub pattern) via the communicator. The UI layer should just have to call one thing.

Reply:
I went through a few options on this one.  That was where I started but I moved away from it.  We need to send the preferences:

 - When saving settings
 - When accepting all
 - When page loads

The event option seems sensible but I'm not sure this is a big enough use-case to justify an event system.